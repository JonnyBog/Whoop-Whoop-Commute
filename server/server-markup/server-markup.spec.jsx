import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';
import configureMockStore from 'redux-mock-store';

import ServerMarkup from 'server-markup/server-markup';

describe('Server', () => {
  describe('Markup', () => {
    it('should export a function', () => {
      expect(ServerMarkup).toBeDefined();
    });

    let props;
    const mockStore = configureMockStore();
    const store = mockStore({});

    beforeEach(() => {
      props = {
        helmetStore: {},
        store,
        url: 'test.com'
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createShallowRenderTree
     * @returns {JSX} shallow renderer component
     */
    function createShallowRenderTree () {
      const shallowRenderer = rendererShallow.createRenderer();

      return shallowRenderer.render(
        <ServerMarkup {...props} />
      );
    }

    it('should return content', () => {
      const tree = createShallowRenderTree();
      expect(tree).toMatchSnapshot();
    });
  });
});
