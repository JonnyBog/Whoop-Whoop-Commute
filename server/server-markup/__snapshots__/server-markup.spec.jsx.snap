// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Markup should return content 1`] = `
<SideEffectProvider(NullComponent)
  store={Object {}}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "600px",
          "1024px",
          "1600px",
        ],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <StaticRouter
        basename=""
        context={Object {}}
        location="test.com"
      >
        <Root />
      </StaticRouter>
    </Provider>
  </ThemeProvider>
</SideEffectProvider(NullComponent)>
`;
